From 648294a02bfcd0eddae51877e3b30f8bbb2d4bb6 Mon Sep 17 00:00:00 2001
From: "Vlastimil Babka (SUSE)" <vbabka@kernel.org>
Date: Thu, 30 Jul 2026 16:32:46 +0200
Subject: [PATCH 10511/15333] mm/slab: stop exporting
 kvfree_rcu_barrier[_on_cache]()

No module code calls these functions directly. Seems it was always the
case. Remove the exports.

Acked-by: Paul E. McKenney <paulmck@kernel.org>
Reviewed-by: Harry Yoo <harry@kernel.org>
Link: https://patch.msgid.link/20260730-unexport-barriers-v1-1-852f6641abe9@kernel.org
Signed-off-by: Vlastimil Babka (SUSE) <vbabka@kernel.org>
---
 mm/slab_common.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/mm/slab_common.c b/mm/slab_common.c
index 39ca4b4a182a..e71a1b346b36 100644
--- a/mm/slab_common.c
+++ b/mm/slab_common.c
@@ -1329,14 +1329,12 @@ void kvfree_rcu_barrier(void)
 	deferred_work_barrier();
 	rcu_barrier();
 }
-EXPORT_SYMBOL_GPL(kvfree_rcu_barrier);
 
 void kvfree_rcu_barrier_on_cache(struct kmem_cache *s)
 {
 	deferred_work_barrier();
 	rcu_barrier();
 }
-EXPORT_SYMBOL_GPL(kvfree_rcu_barrier_on_cache);
 
 void __init kvfree_rcu_init(void)
 {
@@ -2163,7 +2161,6 @@ void kvfree_rcu_barrier(void)
 	flush_all_rcu_sheaves();
 	__kvfree_rcu_barrier();
 }
-EXPORT_SYMBOL_GPL(kvfree_rcu_barrier);
 
 /**
  * kvfree_rcu_barrier_on_cache - Wait for in-flight kvfree_rcu() calls on a
@@ -2186,7 +2183,6 @@ void kvfree_rcu_barrier_on_cache(struct kmem_cache *s)
 	rcu_barrier();
 	__kvfree_rcu_barrier();
 }
-EXPORT_SYMBOL_GPL(kvfree_rcu_barrier_on_cache);
 
 static unsigned long
 kfree_rcu_shrink_count(struct shrinker *shrink, struct shrink_control *sc)
-- 
2.55.0

